src/App.tsx

```diff
@@ -1,13 +1,74 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+
+type Filter = 'All' | 'Active' | 'Completed';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [selectedTodo] = useState<Todo | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState<Filter>('All');
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage(`Unable to load todos`))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  //#region functions
+  useEffect(() => {
+    if (errorMessage.length !== 0) {
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, [errorMessage]);
+
+  function getVisibleTodos() {
+    let visibleTodos: Todo[];
+
+    switch (selectedFilter) {
+      case 'All':
+        visibleTodos = [...todos];
+        break;
+      case 'Active':
+        visibleTodos = [...todos].filter(todo => !todo.completed);
```

Comment:
```suggestion
        visibleTodos = todos.filter(todo => !todo.completed);
```
the filter method doesn't mutate the array